### PR TITLE
Switch to .exec() variants of mongoose operations

### LINKED ISSUE #122: Switch to .exec() variants of mongoose operations
We unnecessarily use Mongoose functions that utilize callbacks instead of promises, which causes nesting.

### PR DESCRIPTION
**Pull request by Jordan141**
Fixes #122 

As with all heavy refactors, needs heavy testing on more than one machine.

### CODE REVIEW

**Comment by Jordan141** [comment]
As this changes many routes, blocking until all other PRs are merged

File: app/routes/articles.js [hunk 1]
```diff
@@ -77,15 +77,18 @@ router.get('/new', isLoggedIn, hasAuthorRole, (req, res) => {
...
         req.flash('Oops! Something went wrong!')
         return res.redirect('/')
     }
-    const listingPageNumber = parseInt(req.query?.page) || 1
-    return articleListingPromise(ALL, {}, req.user.isAdmin).
-        then(articles => res.render('pages/approve', {title: 'Approve Articles', articles, currentUser: req.user, categories: CATEGORIES_LIST, listingPageNumber, isReviewing: true})).
-        catch(err => res.render('error', {code: 500, msg: err}))
+  const listingPageNumber = parseInt(req.query?.page) || 1
+    try {
+        const articles = articleListingPromise(ALL, {}, req.user.isAdmin).exec()
```
Comment (michaelKurowski) [inline app/routes/articles.js:87]: You likely wanted to use await here

Comment (michaelKurowski) [reply app/routes/articles.js:87]: otherwise it won't work, you'll see JSON with promise inside

File: app/routes/articles.js [hunk 2]
```diff
@@ -107,33 +110,26 @@ router.get('/approve/:link', isLoggedIn, async (req, res) => {
...
+router.post('/approve/:link', isLoggedIn, async (req, res) => {
     if(!req.user.isAdmin || !req.params.link) {
         req.flash('error', 'Oops! Something went wrong!')
         return res.redirect('/')
     }
 
-    Article.findOne({link: req.params.link}, (err, article) => {
-        if(err) return res.sendStatus(500)
+    try {
+        const article = Article.findOne({link: req.params.link}).populate('author').exec()
```
Comment (michaelKurowski) [inline app/routes/articles.js:120]: await ?

File: app/routes/articles.js [hunk 3]
```diff
@@ -198,62 +194,65 @@ async function getRecommendedArticles(category) {
...
     if(!req.params.link) return res.render('error', {code: '404', msg: 'Invalid Article Link'})
-   
-    Article.deleteOne({link: req.params.link}, err => {
-        if(err) return res.render('error', {code: '500', msg: 'Internal Database Error'})
-
-        Counter.deleteOne({articleLink: req.params.link}).exec()
+    
+    try {
+        await Article.deleteOne({link: req.params.link}).exec()
+        await Counter.deleteOne({articleLink: req.params.link}).exec()
```
Comment (michaelKurowski) [inline app/routes/articles.js:250]: Is it possible for two articles to have the same link?
i.e  articles with titles `How to hack node` and `How to "hack" node`

**Review by michaelKurowski — CHANGES_REQUESTED** [review]

File: app/routes/articles.js (hunk 3 above)
Comment (Jordan141) [reply app/routes/articles.js:251]: Article titles are alphanumeric that allow dashes and spaces, it shouldn't be possible, link must be unique.

**Review by michaelKurowski — APPROVED** [review]

**Pull request merged**